### PR TITLE
MANTA-4769 joyent/buckets-mdapi could do with a round of formatting

### PR DESCRIPTION
**Pull request by chudley**
I've also included TOOLS-2377 (standardize on "./rust-toolchain" to explicitly lock rust version in release repos) into this commit, but I'm happy to remove it if we still want further discussion on that.

**Testing**

Ran `cargo run --bin=boray  -- -c config.toml | bunyan` against a local database and performed some `fastcall`s.

Running/compiling "boray" didn't complain, and the server started just fine and would accept and respond to my `fastcall`s. It seems that "schema-manager" looks for some Manta-specific paths to config, and so I only checked that it compiled OK with `cargo run --bin=schema-manager` (it tried to run `mdata-get SAPI_URL`, which should show that there are at least no compilation issues).

Both of the above steps were performed using the new `rust-toolchain` in place, which appeared to correctly choose the right version I had locally installed. I ran `rustup toolchain uninstall 1.33.0-x86_64-apple-darwin` and then attempted another build of the application, at which point `rustup` appeared to download the correct toolchain for me and build the application. `rustup show` also showed that the 1.33.0 toolchain was active. This didn't seem to have an impact on our Jenkins build, and I deployed that image (d2d88860-84a0-43ca-a4cd-a03654ac5c80) to my lab with success, but I don't know how this impacts *development* on a SmartOS machine.

### CODE REVIEW

File: schema-manager/src/main.rs [hunk 1]
```diff
@@ -223,27 +235,27 @@ fn run(log: &Logger) -> Result<(), Box<dyn std::error::Error>>
...
+    let eb_endpoint = [eb_address, String::from(":"), DEFAULT_EB_PORT.to_string()].concat();
     info!(log, "electric-boray endpoint:{}", eb_endpoint);
     let mut stream = TcpStream::connect(&eb_endpoint).unwrap_or_else(|e| {
         error!(log, "failed to connect to electric-boray: {}", e);
         process::exit(1)
     });
 
     let mut msg_id = FastMessageId::new();
-    let recv_cb = |msg: &FastMessage| { vnode_response_handler(&log, msg) };
+    let recv_cb = |msg: &FastMessage| vnode_response_handler(&log, msg);
```
Comment (bowrocker) [inline schema-manager/src/main.rs]: Not sure about this change to the closure, I believe I tried it this way and it didn't work. I will test it out.

Comment (bowrocker) [reply schema-manager/src/main.rs]: Changes on schema-manager look good:
```
[root@66f23311 (boray) /opt/smartdc/boray/bin]$ ./schema-manager
INFO: Running "/usr/sbin/mdata-get SAPI_URL" ...
Nov 26 18:31:30.944 INFO sapi_url:http://sapi.jonran-dc.joyent.us, build-id: 0.1.0
INFO: Running "/usr/bin/zonename" ...
Nov 26 18:31:31.196 INFO pnode argument to electric-boray:tcp://1.boray.jonran-region.joyent.us:2030, build-id: 0.1.0
Nov 26 18:31:31.196 INFO electric-boray endpoint:electric-boray.jonran-region.joyent.us:2020, build-id: 0.1.0
Nov 26 18:31:31.225 INFO processing vnode: "0", build-id: 0.1.0
Nov 26 18:31:31.228 INFO Creating boray user and role if they don't exist on postgres://postgres:postgres@10.77.77.22:5432, build-id: 0.1.0
Nov 26 18:31:31.263 INFO Creating boray database if it doesn't exist on postgres://postgres:postgres@10.77.77.22:5432, build-id: 0.1.0
Nov 26 18:31:31.305 INFO Creating boray schemas on postgres://boray:boray@10.77.77.22:5432/boray, build-id: 0.1.0
Nov 26 18:31:31.310 INFO processing vnode: "2", build-id: 0.1.0
Nov 26 18:31:31.310 INFO Creating boray user and role if they don't exist on postgres://postgres:postgres@10.77.77.22:5432, build-id: 0.1.0
Nov 26 18:31:31.329 INFO Creating boray database if it doesn't exist on postgres://postgres:postgres@10.77.77.22:5432, build-id: 0.1.0
Nov 26 18:31:31.432 INFO Creating boray schemas on postgres://boray:boray@10.77.77.22:5432/boray, build-id: 0.1.0
Nov 26 18:31:31.436 INFO Done., build-id: 0.1.0
```

**Comment by chudley** [comment]
Nice, thanks for checking that out!

**Comment by chudley** [comment]
Sorry for the noise, but I think it's best not to include TOOLS-2377 yet and to just address the formatting in this change. I was reading the ticket just before merging, and it seems for SmartOS platforms there's actually still a bit of work to do in https://github.com/joyent/eng. I don't know how many of us develop directly on SmartOS, but probably best to not rock any boats there for now.

**Comment by chudley** [comment]
2 commits have gone in since I planned to `cargo fmt` and both have caused conflicts that needed resolving. I'm going to close this for now because 1) it's going to be cleaner to not have to deal with merge commits here, and 2) I need to figure out how to run boray for testing with the manatee state changes.

**Pull request closed without merging**